crates/task-impls/src/quorum_vote.rs

```diff
@@ -0,0 +1,383 @@
+        &self,
+        dependency_type: VoteDependency,
+        view_number: TYPES::Time,
+        event_receiver: Receiver<Arc<HotShotEvent<TYPES>>>,
+    ) -> EventDependency<Arc<HotShotEvent<TYPES>>> {
+        EventDependency::new(
+            event_receiver.clone(),
+            Box::new(move |event| {
+                let event = event.as_ref();
+                let event_view = match dependency_type {
+                    VoteDependency::QuorumProposal => {
+                        if let HotShotEvent::QuorumProposalValidated(proposal) = event {
+                            proposal.view_number
+                        } else {
+                            return false;
+                        }
+                    }
+                    VoteDependency::Dac => {
+                        if let HotShotEvent::DACValidated(cert) = event {
+                            cert.view_number
+                        } else {
+                            return false;
+                        }
+                    }
+                    VoteDependency::Vid => {
+                        if let HotShotEvent::VidDisperseValidated(disperse) = event {
+                            disperse.view_number
+                        } else {
+                            return false;
+                        }
+                    }
+                };
+                event_view == view_number
+            }),
+        )
+    }
+
+    /// Create and store an [`AndDependency`] combining [`EventDependency`]s associated with the
+    /// given view number if it doesn't exist.
+    fn create_dependency_task_if_new(
+        &mut self,
+        view_number: TYPES::Time,
+        event_receiver: Receiver<Arc<HotShotEvent<TYPES>>>,
+        event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
+    ) {
+        if self.vote_dependencies.get(&view_number).is_some() {
+            return;
+        }
+        let deps = vec![
+            self.create_event_dependency(
+                VoteDependency::QuorumProposal,
+                view_number,
+                event_receiver.clone(),
+            ),
+            self.create_event_dependency(VoteDependency::Dac, view_number, event_receiver.clone()),
+            self.create_event_dependency(VoteDependency::Vid, view_number, event_receiver),
+        ];
+        let vote_dependency = AndDependency::from_deps(deps);
+        let dependency_task = DependencyTask::new(
+            vote_dependency,
+            VoteDependencyHandle {
+                view_number,
+                sender: event_sender,
+            },
+        );
+        self.vote_dependencies.insert(
+            view_number,
+            async_spawn(async move {
+                #[cfg(async_executor_impl = "async-std")]
+                dependency_task.run().await;
+                // Allow `unwrap()` for tokio.
+                #[cfg(async_executor_impl = "tokio")]
+                dependency_task.run().await.unwrap();
+            }),
+        );
```

Comment:
I think this can just call run on the task.  It internally does the spawning.
```suggestion
        self.vote_dependencies.insert(
            view_number,
            dependency_task.run(),
        );
```